src/System.Runtime.WindowsRuntime/tests/System/IO/AsWinRTStreamTests.cs

```diff
@@ -0,0 +1,413 @@
+            Stream managedStream = TestStreamProvider.CreateWriteOnlyStream();
+            using (IOutputStream outs = managedStream.AsOutputStream())
+            {
+                Assert.Same(managedStream, outs.AsStreamForWrite());
+            }
+        }
+
+        [Fact]
+        public static void AsInputStream_Equal()
+        {
+            Stream stream = TestStreamProvider.CreateReadOnlyStream();
+
+            using (IInputStream insOne = stream.AsInputStream())
+            using (IInputStream insTwo = stream.AsInputStream())
+            {
+                Assert.Equal(insOne, insTwo);
+            }
+        }
+
+        [Fact]
+        public static void AsInputStream_NotEqual()
+        {
+            Stream streamOne = TestStreamProvider.CreateReadOnlyStream();
+            Stream streamTwo = TestStreamProvider.CreateReadOnlyStream();
+
+            Assert.NotEqual(streamOne, streamTwo);
+
+            using (IInputStream insOne = streamOne.AsInputStream())
+            using (IInputStream insTwo = streamTwo.AsInputStream())
+            {
+                Assert.NotEqual(insOne, insTwo);
+            }
+        }
+
+        [Fact]
+        public static void AsOutputStream_Equal()
+        {
+            Stream stream = TestStreamProvider.CreateWriteOnlyStream();
+
+            using (IOutputStream outsOne = stream.AsOutputStream())
+            using (IOutputStream outsTwo = stream.AsOutputStream())
+            {
+                Assert.Equal(outsOne, outsOne);
+            }
+        }
+
+        [Fact]
+        public static void AsOutputStream_NotEqual()
+        {
+            Stream streamOne = TestStreamProvider.CreateWriteOnlyStream();
+            Stream streamTwo = TestStreamProvider.CreateWriteOnlyStream();
+
+            Assert.NotEqual(streamOne, streamTwo);
+
+            using (IOutputStream outsOne = streamOne.AsOutputStream())
+            using (IOutputStream outsTwo = streamTwo.AsOutputStream())
+            {
+                Assert.NotEqual(outsOne, outsTwo);
+            }
+        }
+
+        [Fact]
+        public static void TestRead_MemoryStream_None()
+        {
+            DoTestRead(() => { return TestStreamProvider.CreateMemoryStreamAsInputStream(); }, InputStreamOptions.None, false, true);
+        }
+
+        [Fact]
+        public static void TestRead_MemoryStream_Partial()
+        {
+            DoTestRead(() => { return TestStreamProvider.CreateMemoryStreamAsInputStream(); }, InputStreamOptions.Partial, false, true);
+        }
+
+        [Fact]
+        public static void TestWrite_MemoryStream()
+        {
+            DoTestWrite(() => { return TestStreamProvider.CreateMemoryStream(); }, false);
+        }
+
+        private static volatile bool s_progressCallbackInvoked = false;
+        private static volatile bool s_completedCallbackInvoked = false;
+        private static volatile IBuffer s_buffer;
+        private static volatile uint s_resultBytesWritten = 0;
```

Comment:
Why are these statics rather than locals that get closed over?  Reasoning about the safety/reliability of the tests is much harder with statics like this.